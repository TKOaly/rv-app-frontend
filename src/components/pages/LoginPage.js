import './styles/LoginPage.scss';
import { Col, Grid, Row } from 'react-flexbox-grid';
import { Redirect } from 'react-router-dom';
import Header from '../sections/Header';
import LoginForm from '../forms/LoginForm';
import React from 'react';

const LoginPage = ({ authenticated }) => {
    if (authenticated) {
        return <Redirect to="/" />;
    }
    return (
        <div className="loginPage">
            <Header />
            <Grid fluid>
                <Row center="xs">
                    <Col xs={5}>
                        <LoginForm />
                    </Col>
                </Row>
            </Grid>
        </div>
    );
};

export default LoginPage;
