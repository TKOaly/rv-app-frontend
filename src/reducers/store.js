import { applyMiddleware, combineReducers, createStore } from 'redux';
import { createBrowserHistory, createMemoryHistory } from 'history';
import { logger } from './middleware';
import thunk from 'redux-thunk';
// Import reducers
import { composeWithDevTools } from 'redux-devtools-extension';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { reducer as formReducer } from 'redux-form';
import authenticationReducer from './authenticationReducer';
import depositReducer from './depositReducer';
import modalReducer from './modalReducer';
import notificationReducer from './notificationReducer';
import productReducer from './productReducer';
import registerReducer from './registerReducer';
import terminalReducer from './terminalReducer';
import userReducer from './userReducer';

// If we are in a test environment, use memory history
export const history = process.env.NODE_ENV === 'test' ? createMemoryHistory() : createBrowserHistory();

// Combine reducers
const rootReducer = combineReducers({
    notification: notificationReducer,
    authentication: authenticationReducer,
    register: registerReducer,
    terminal: terminalReducer,
    user: userReducer,
    products: productReducer,
    deposit: depositReducer,
    modal: modalReducer,
    form: formReducer,
    router: connectRouter(history)
});

const router = routerMiddleware(history);
const middleware = process.env.NODE_ENV !== 'production' ? [thunk, router, logger] : [thunk, router];

// Create store
export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middleware)));
