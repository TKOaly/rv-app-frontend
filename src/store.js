// Import reducers
import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { reducer as formReducer } from 'redux-form';
import { logger } from './reducers/middleware';
import authenticationReducer, { authenticationActions } from './reducers/authenticationReducer';
import depositReducer from './reducers/depositReducer';
import modalReducer from './reducers/modalReducer';
import notificationReducer from './reducers/notificationReducer';
import productReducer from './reducers/productReducer';
import registerReducer from './reducers/registerReducer';
import terminalReducer from './reducers/terminalReducer';
import thunk from 'redux-thunk';
import userReducer from './reducers/userReducer';

// Combine reducers
const reducer = combineReducers({
    notification: notificationReducer,
    authentication: authenticationReducer,
    register: registerReducer,
    terminal: terminalReducer,
    user: userReducer,
    products: productReducer,
    deposit: depositReducer,
    modal: modalReducer,
    form: formReducer
});

const rootReducer = (state, action) => {
    if (action.type === authenticationActions.LOGOUT_SUCCESS) {
        state = undefined;
    }

    return reducer(state, action);
};

const middleware = process.env.NODE_ENV !== 'production' ? [thunk, logger] : [thunk];

// Create store
export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middleware)));
