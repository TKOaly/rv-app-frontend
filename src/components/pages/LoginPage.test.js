import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configure, shallow } from 'enzyme';
import { store } from '../../store';
import Adapter from 'enzyme-adapter-react-16';
import LoginPage from './LoginPage';
import React from 'react';

configure({ adapter: new Adapter() });

/*
Here we pass in the Redux store and a memory router,
 to test the functionality of the login page.
*/

describe('<LoginPage />', () => {
    it('renders correctly when not authenticated', () => {
        const cmpnt = shallow(
            <Provider store={store}>
                <MemoryRouter>
                    <LoginPage authenticated={false} />
                </MemoryRouter>
            </Provider>
        );
        expect(cmpnt).toMatchSnapshot();
    });
    it('renders correctly when authenticated', () => {
        const cmpnt = shallow(
            <Provider store={store}>
                <MemoryRouter>
                    <LoginPage authenticated={true} />
                </MemoryRouter>
            </Provider>
        );
        expect(cmpnt).toMatchSnapshot();
    });
});
