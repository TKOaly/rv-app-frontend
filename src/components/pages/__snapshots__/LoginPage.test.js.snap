// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders correctly when authenticated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter>
      <LoginPage
        authenticated={true}
      />
    </MemoryRouter>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <MemoryRouter>
        <LoginPage
          authenticated={true}
        />
      </MemoryRouter>,
      "value": Object {
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeState": Object {
          "authentication": Object {
            "access_token": "",
            "isLoggingIn": false,
            "loggedIn": false,
          },
          "deposit": Object {
            "confirmationVisibility": false,
            "depositAmountText": "",
          },
          "form": Object {},
          "modal": Object {
            "modalContent": null,
            "modalVisible": false,
            "props": null,
          },
          "notification": Object {
            "notifications": Array [],
            "purchaseNotificationStartTime": null,
            "purchaseNotificationTimeout": 2500,
            "purchasedItems": Array [],
          },
          "products": Object {
            "buyAmount": 1,
            "categories": Array [],
            "filter": "",
            "gettingCategories": false,
            "gettingProducts": false,
            "products": Array [],
            "selectedCategory": -1,
          },
          "register": Object {
            "registering": false,
          },
          "terminal": Object {
            "inputValid": false,
            "terminalInput": "",
          },
          "user": Object {
            "account_balance": 0,
            "email": "",
            "full_name": "",
            "username": "",
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <LoginPage
          authenticated={true}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authenticated": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <MemoryRouter>
          <LoginPage
            authenticated={true}
          />
        </MemoryRouter>,
        "value": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "storeState": Object {
            "authentication": Object {
              "access_token": "",
              "isLoggingIn": false,
              "loggedIn": false,
            },
            "deposit": Object {
              "confirmationVisibility": false,
              "depositAmountText": "",
            },
            "form": Object {},
            "modal": Object {
              "modalContent": null,
              "modalVisible": false,
              "props": null,
            },
            "notification": Object {
              "notifications": Array [],
              "purchaseNotificationStartTime": null,
              "purchaseNotificationTimeout": 2500,
              "purchasedItems": Array [],
            },
            "products": Object {
              "buyAmount": 1,
              "categories": Array [],
              "filter": "",
              "gettingCategories": false,
              "gettingProducts": false,
              "products": Array [],
              "selectedCategory": -1,
            },
            "register": Object {
              "registering": false,
            },
            "terminal": Object {
              "inputValid": false,
              "terminalInput": "",
            },
            "user": Object {
              "account_balance": 0,
              "email": "",
              "full_name": "",
              "username": "",
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <LoginPage
            authenticated={true}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authenticated": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<LoginPage /> renders correctly when not authenticated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter>
      <LoginPage
        authenticated={false}
      />
    </MemoryRouter>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <MemoryRouter>
        <LoginPage
          authenticated={false}
        />
      </MemoryRouter>,
      "value": Object {
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeState": Object {
          "authentication": Object {
            "access_token": "",
            "isLoggingIn": false,
            "loggedIn": false,
          },
          "deposit": Object {
            "confirmationVisibility": false,
            "depositAmountText": "",
          },
          "form": Object {},
          "modal": Object {
            "modalContent": null,
            "modalVisible": false,
            "props": null,
          },
          "notification": Object {
            "notifications": Array [],
            "purchaseNotificationStartTime": null,
            "purchaseNotificationTimeout": 2500,
            "purchasedItems": Array [],
          },
          "products": Object {
            "buyAmount": 1,
            "categories": Array [],
            "filter": "",
            "gettingCategories": false,
            "gettingProducts": false,
            "products": Array [],
            "selectedCategory": -1,
          },
          "register": Object {
            "registering": false,
          },
          "terminal": Object {
            "inputValid": false,
            "terminalInput": "",
          },
          "user": Object {
            "account_balance": 0,
            "email": "",
            "full_name": "",
            "username": "",
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <LoginPage
          authenticated={false}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authenticated": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <MemoryRouter>
          <LoginPage
            authenticated={false}
          />
        </MemoryRouter>,
        "value": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "storeState": Object {
            "authentication": Object {
              "access_token": "",
              "isLoggingIn": false,
              "loggedIn": false,
            },
            "deposit": Object {
              "confirmationVisibility": false,
              "depositAmountText": "",
            },
            "form": Object {},
            "modal": Object {
              "modalContent": null,
              "modalVisible": false,
              "props": null,
            },
            "notification": Object {
              "notifications": Array [],
              "purchaseNotificationStartTime": null,
              "purchaseNotificationTimeout": 2500,
              "purchasedItems": Array [],
            },
            "products": Object {
              "buyAmount": 1,
              "categories": Array [],
              "filter": "",
              "gettingCategories": false,
              "gettingProducts": false,
              "products": Array [],
              "selectedCategory": -1,
            },
            "register": Object {
              "registering": false,
            },
            "terminal": Object {
              "inputValid": false,
              "terminalInput": "",
            },
            "user": Object {
              "account_balance": 0,
              "email": "",
              "full_name": "",
              "username": "",
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <LoginPage
            authenticated={false}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authenticated": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
